server/claimRevision/schema/claim-revision.schema.ts

```diff
@@ -0,0 +1,43 @@
+import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
+import * as mongoose from "mongoose";
+import { Claim } from "../../claim/schemas/claim.schema";
+import { Source } from "../../source/schemas/source.schema";
+
+export type ClaimRevisionDocument = ClaimRevision & mongoose.Document;
+
+@Schema({ toObject: {virtuals: true}, toJSON: {virtuals: true} })
+export class ClaimRevision {
+    @Prop({ required: true })
+    title: string;
+
+    @Prop({ required: true })
+    slug: string;
+
+    @Prop({ type: Object, required: true })
+    content: object;
+
+    @Prop({ required: true })
+    date: Date;
+
+    @Prop({ required: true })
+    claim: Claim;
+
+    @Prop({ required: true })
+    revisionId: number;
+}
+
+const ClaimRevisionSchemaRaw = SchemaFactory.createForClass(ClaimRevision);;
+
+ClaimRevisionSchemaRaw.virtual('claims', {
```

Comment:
Not needed, one revision has only one Claim.

In the Claim schema however, we will be needing a populate virtual for revisions:
```
ClaimSchemaRaw.virtual('revisions', {
  ref: 'ClaimRevision',
  localField: '_id',
  foreignField: 'claimId'
})
```